File: packages/pyre/calc/Node.py
# -*- Python -*-
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2026 all rights reserved


# metaclass
from . import calculator


# declaration of the base node
class Node(metaclass=calculator):
    """
    The base class for lazily evaluated nodes
    """

    # not much to do


# end of file

File: lib/pyre/flow/Node.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved

// externals
#include "external.h"
// forward declarations
#include "forward.h"
// type aliases
#include "api.h"

// my class declaration
#include "Node.h"

// internals
auto
pyre::flow::Node::flush() -> void
{
    // mark me as stale
    _stale = true;
    // all done
    return;
}

// end of file